app/Role.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace App;
+
+use Illuminate\Database\Eloquent\Model;
+
+class Role extends Model
+{
+    protected $fillable = [
+        'name',
+    ];
+
+    public function users()
+    {
+        return $this->belongsToMany(User::class, 'role_users');
```

Comment:
Missing withPivot('object_id').

Reply:
Also at the reverse of this relation in the User class that I forgot to mark.

Reply:
Thanks! I did not know about withPivot.